fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/dao/impl/ResourceDAOImpl.java

```diff
@@ -535,7 +541,16 @@ public Resource insert(Resource resource, List<ExtractedParameterValue> paramete
             stmt = connection.prepareCall(stmtString);
             stmt.setString(1, resource.getResourceType());
             stmt.setString(2, resource.getLogicalId());
-            stmt.setBytes(3, resource.getData());
+
+            boolean large = false;
+            if (SIZE_LIMIT < resource.getData().length) {
+                // Outside of the normal flow we have a BIG JSON or XML
+                stmt.setNull(3, java.sql.Types.NULL);
```

Comment:
You should specify the actual data-type here: java.sql.Types.BLOB.

Reply:
Done